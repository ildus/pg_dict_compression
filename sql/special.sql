CREATE SCHEMA test;
CREATE EXTENSION pg_dict_compression SCHEMA test;
CREATE TABLE test.c1(
	a BYTEA COMPRESSION pg_dict_compression WITH (dict 'german he she her budget long')
);

CREATE OR REPLACE FUNCTION test.dict_compression_test(OID, BYTEA)
RETURNS BYTEA
AS 'pg_dict_compression', 'dict_compression_test'
LANGUAGE C STRICT;

CREATE OR REPLACE FUNCTION test.dict_compression_test_wrapper(input BYTEA)
RETURNS BYTEA AS $$
DECLARE
	cmid OID;
	res  BYTEA;
BEGIN
	SELECT acoid FROM pg_attr_compression INTO cmid WHERE acrelid = 'test.c1'::regclass;
	res := test.dict_compression_test(cmid, input);
	RETURN res;
END
$$ LANGUAGE plpgsql;

SELECT test.dict_compression_test_wrapper(E'\\x6865FF736865'); /* 6 */
SELECT test.dict_compression_test_wrapper(E'\\x627564676572FF6d616e');	/* null */
SELECT test.dict_compression_test_wrapper(E'\\x627564676572FF6765726d616e');	/* 10 */

DROP SCHEMA test CASCADE;
