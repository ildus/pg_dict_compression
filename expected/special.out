CREATE SCHEMA test;
CREATE EXTENSION pg_dict_compression SCHEMA test;
CREATE TABLE test.c1(
	a BYTEA COMPRESSION pg_dict_compression WITH (dict 'german he she her budget long')
);
CREATE OR REPLACE FUNCTION test.dict_compression_test(OID, BYTEA)
RETURNS BYTEA
AS 'pg_dict_compression', 'dict_compression_test'
LANGUAGE C STRICT;
CREATE OR REPLACE FUNCTION test.dict_compression_test_wrapper(input BYTEA)
RETURNS BYTEA AS $$
DECLARE
	cmid OID;
	res  BYTEA;
BEGIN
	SELECT acoid FROM pg_attr_compression INTO cmid WHERE acrelid = 'test.c1'::regclass;
	res := test.dict_compression_test(cmid, input);
	RETURN res;
END
$$ LANGUAGE plpgsql;
SELECT test.dict_compression_test_wrapper(E'\\x6865FF736865'); /* 6 */
NOTICE:  len: 6
 dict_compression_test_wrapper 
-------------------------------
 \x6865ff736865
(1 row)

SELECT test.dict_compression_test_wrapper(E'\\x627564676572FF6d616e');	/* null */
ERROR:  res is null
CONTEXT:  PL/pgSQL function test.dict_compression_test_wrapper(bytea) line 7 at assignment
SELECT test.dict_compression_test_wrapper(E'\\x627564676572FF6765726d616e');	/* 14 */
NOTICE:  len: 10
 dict_compression_test_wrapper 
-------------------------------
 \x627564676572ff6765726d616e
(1 row)

DROP SCHEMA test CASCADE;
NOTICE:  drop cascades to 4 other objects
DETAIL:  drop cascades to extension pg_dict_compression
drop cascades to table test.c1
drop cascades to function test.dict_compression_test(oid,bytea)
drop cascades to function test.dict_compression_test_wrapper(bytea)
